pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1321,6 +1321,11 @@ type ObjectStoreSpec struct {
 	// +optional
 	// +nullable
 	Security *SecuritySpec `json:"security,omitempty"`
+
+	// +optional
+	// +kubebuilder:validation:Enum={"kms","s3","kms|s3"}
+	// Defines the aws-sse supported for RGW https://docs.aws.amazon.com/AmazonS3/latest/userguide/serv-side-encryption.html
+	AWSServerSideEncryptionType string `json:"awsServerSideEncryptionType,omitempty"`
```

Comment:
Why is this called `AWS...`? If Ceph is implementing this, then it's no longer specific to AWS right? Why not just call this `ServerSideEncryptionType`?

Reply:
@BlaineEXE : This `ServerSIdeEncryption` is specific to s3 protocol or AWS hence prefixed it with AWS. For RGW itself to support a `serverside encryption` or supports for `swift` then it option won't be applicable.
So from the above discussions I guess struct may look like
`ServerSideEncryption {
          AWS {
                    KMSEnabled
                    S3Enabled
}
}`